### PR TITLE
Add topics regarding Network Security Config

### PR DESCRIPTION
**Pull request by clviper**
This two topics cover two security issues regarding Network Security Config.

- When apps are use a custom Network Security Configuration file to override the default Network Security Configuration present in devices with Android 7+ to allow trust in user supplied CA's.

- When apps still don't target API levels 24+, meaning that when running on devices with Android 7+, still don't take advantage of this protection.

### CODE REVIEW

File: Document/0x05g-Testing-Network-Communication.md [hunk 1]
```diff
@@ -242,6 +242,53 @@ For further information, please check the [OWASP certificate pinning guide](http
...
+<?xml version="1.0" encoding="utf-8"?>
+<network-security-config>
+    <base-config>
+        <trust-anchors>
+            <certificates src="system"/>
+            <certificates src="user"/>
+        </trust-anchors>
+    </base-config>
+</network-security-config>
+```
```
[the remark this answers is not in the document]
Comment (clviper) [reply Document/0x05g-Testing-Network-Communication.md]: The problem is that the one used for the pinning does not have a <trust anchor> and a <certificates> entry.

File: Document/0x05g-Testing-Network-Communication.md [hunk 2]
```diff
@@ -242,6 +242,53 @@ For further information, please check the [OWASP certificate pinning guide](http
...
+
+When apps target API Levels 24+ and are running on an Android device with versions 7+, they used a default Network Security Configuration that doest not trust user supplied CA's, reducing the possibility of MiTM attacks by luring users to install malicious CA's.
+
+This protection can be bypassed by using a custom Network Security Configuration with a custom trust anchor indicating that the app will trust user supplied CA's.
+
+#### Static Analysis
+
+The Network Security Configuration should be analysed to determine what settings are configured. The file is located inside the apk in the /res/xml/ folder with the name network_security_config.xml.
+
+If there are custom <trust-anchors> present in a <base-config> or <domain-config>, that define a <certificates src="user"> the application will trust user supplied CA's for those particular domains or for all domains. Example:
```
[the remark this answers is not in the document]
Comment (clviper) [reply Document/0x05g-Testing-Network-Communication.md]: Those are the root elements possible to define in a Network Security Configuration. <base-config>, <domain-config> and also <debug-overrides>.

File: Document/0x05g-Testing-Network-Communication.md [hunk 3]
```diff
@@ -242,6 +242,53 @@ For further information, please check the [OWASP certificate pinning guide](http
...
+
+#### Overview
+As mentioned in the previous topic, apps that target API levels 24+, unless otherwise defined, will implement a default Network Security Configuration that no longer trust user supplied CA's.
+
+In a scenario that the app is running on a Android device with version 7+, but targets API levels below 24, it will not use this feature, therefore still trusting in user supplied CA's.
+
+#### Static Analysis
+
+* Use a decompiler (Ex. Jadx) to confirm the targetSDK present in the AndroidManifest.xml file
+* Use apktool to decode the app and confirm the targetSDK present in the file apktool.yml of the output folder.
```
[the remark this answers is not in the document]
Comment (clviper) [reply Document/0x05g-Testing-Network-Communication.md]: Sure.

File: Document/0x05g-Testing-Network-Communication.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (commjoen) [reply Document/0x05g-Testing-Network-Communication.md]: Good point! Let's keep it this way then :)

File: Document/0x05g-Testing-Network-Communication.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (commjoen) [reply Document/0x05g-Testing-Network-Communication.md]: Ah yes, I misread it hte first time, would it be a good idea to have a domain-config example as well?

[the remark this answers is not in the document]
Comment (clviper) [reply Document/0x05g-Testing-Network-Communication.md]: Yup, I'll include it.

**Pull request merged**